orderbook/src/database/trades.rs

```diff
@@ -115,19 +127,23 @@ mod tests {
         return (owners, order_ids);
     }
 
-    async fn add_trade(db: &Database, owner: H160, order_uid: OrderUid, log_index: u64) -> Trade {
+    async fn add_trade(
+        db: &Database,
+        owner: H160,
+        order_uid: OrderUid,
+        event_index: EventIndex,
+        tx_hash: Option<H256>,
```

Comment:
Yes, this is all a bit counter intuitive at this point. Although there are now so many combinations to go along with it. We would have `add_trade`, `add_order_and_trade`, `add_order_trade_and_settlement`, `add_trade_and_settlement`, possibly more.  I will try and come up with something a bit cleaner here.

Reply:
I have made the unit tests more readable/less confusing by adding `add_settlement`. Anything other than this (i.e. combined helpers like add_trade_and_settlement`) didn’t make sense since they are very much independent tasks.